fairseq/utils.py

```diff
@@ -146,13 +146,16 @@ def load_ensemble_for_inference(filenames, task, model_arg_overrides=None):
         state = torch.load(filename, map_location=lambda s, l: default_restore_location(s, 'cpu'))
         state = _upgrade_state_dict(state)
         states.append(state)
-    args = states[0]['args']
+    args_list = []
+    for state in states:
+        args_list.append(state['args'])
     if model_arg_overrides is not None:
-        args = _override_model_args(args, model_arg_overrides)
+        for args in args_list:
+            args = _override_model_args(args, model_arg_overrides)
 
     # build ensemble
     ensemble = []
-    for state in states:
+    for args, state in zip(args_list, states):
```

Comment:
Replace with:
```
for state in states:
    args = state['args']
    if model_arg_overrides is not None:
        args = _override_model_args(state['args'], model_arg_overrides)
    model = task.build_model(args)
    (...)
```

Reply:
Great idea. I will make the changes accordingly. Thanks!